lib/ansible/modules/cloud/docker/docker_swarm_facts.py

```diff
@@ -0,0 +1,319 @@
+      - See L(the docker documentation,https://docs.docker.com/engine/reference/commandline/node_ls/#filtering)
+        for more information on possible filters.
+    type: dict
+  services:
+    description:
+      - Whether to list swarm services.
+    type: bool
+    default: no
+  services_filters:
+    description:
+      - A dictionary of filter values used for selecting services to list.
+      - "For example, C(name: myservice)."
+      - See L(the docker documentation,https://docs.docker.com/engine/reference/commandline/service_ls/#filtering)
+        for more information on possible filters.
+    type: dict
+  tasks:
+    description:
+      - Whether to list containers.
+    type: bool
+    default: no
+  tasks_filters:
+    description:
+      - A dictionary of filter values used for selecting tasks to list.
+      - "For example, C(node: mynode-1)."
+      - See L(the docker documentation,https://docs.docker.com/engine/reference/commandline/service_ps/#filtering)
+        for more information on possible filters.
+    type: dict
+  verbose_output:
+    description:
+      - When set to C(yes) and I(nodes), I(services) or I(tasks) is set to C(yes)
+        then output will contain verbose information about objects matching the full output of API method.
+        For details see the documentation of your version of Docker API at L(https://docs.docker.com/engine/api/).
+      - The verbose output in this module contains only subset of information returned by I(_facts) module
+        for each type of the objects.
+    type: bool
+    default: no
+extends_documentation_fragment:
+    - docker
+
+requirements:
+    - "python >= 2.6"
+    - "docker-py >= 1.10.0"
+    - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+       module has been superseded by L(docker,https://pypi.org/project/docker/)
+       (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+       For Python 2.6, C(docker-py) must be used. Otherwise, it is recommended to
+       install the C(docker) Python module. Note that both modules should I(not)
+       be installed at the same time. Also note that when both modules are installed
+       and one of them is uninstalled, the other might no longer function and a
+       reinstall of it is required."
+    - "Docker API >= 1.24"
+'''
+
+EXAMPLES = '''
+- name: Get info on Docker Swarm
+  docker_swarm_facts:
+  register: result
+
+- name: Get info on Docker Swarm and list of registered nodes
+  docker_swarm_facts:
+    nodes: yes
+  register: result
+
+- name: Get info on Docker Swarm and extended list of registered nodes
+  docker_swarm_facts:
+    nodes: yes
+    verbose_output: yes
+  register: result
+
+- name: Get info on Docker Swarm and filtered list of registered nodes
+  docker_swarm_facts:
+    nodes: yes
+    nodes_filter:
+      name: mynode
+  register: result
+
+- debug:
+    var: result.docker_swarm_facts
+'''
+
+RETURN = '''
+docker_swarm_facts:
```

Comment:
How about removing the `docker_` prefix from the return variables, and simplifying the names in general? I would use:
- `swarm_facts` or just `swarm`
- `nodes`
- `services`
- `tasks`

Reply:
That will break the naming convention of returned values, I am not sure if this is a good idea. Most `docker_*_facts` modules returns variables with `docker_` prefix. Maybe we should open discussion in agenda and unify naming across modules before 2.8 freeze?

Reply:
Funny thing, the only module which was there already in Ansible 2.7 is `docker_image_facts`, and its return value is called `images` :-) Let's change it in all `_facts` modules, then. I'll create a PR for that.

Reply:
I am not so sure it is a good approach. I know there is no standard across modules, but it doesn't hurt to have the prefix and it may be useful if you do some manipulations from multiple `_facts` modules from different clouds.

Reply:
See #51939.

Reply:
I disagree: if you (as the module's user) want to make sure that you have something around which reminds you of docker, why not put it into the `register:` variable's name then?

Reply:
Also, let's continue the discussion in #51939.